frontend/ui/widget/imageviewer.lua

```diff
@@ -621,7 +638,7 @@ end
 function ImageViewer:onZoomIn(inc)
     if self.scale_factor == 0 then
         -- Get the scale_factor made out for best fit
-        self.scale_factor = self._image_wg:getScaleFactor()
+        self.scale_factor = self._scale_factor_0 or self._image_wg:getScaleFactor()
     end
     if not inc then inc = 0.2 end -- default for key zoom event
     self.scale_factor = self.scale_factor + inc
```

Comment:
At a first glance the following lines look a bit odd; they can be optimized. Dunno if that solves the problem.
But I have to dig deeper into this widget (as I have never touched it before).

Reply:
Yes, the frustration is probably around these lines 643-644. It's possible I had tried multiplication instead of addition (and now that you say it, adding 0.2 to the scale factor will indeed make zoom-in slow :) and was fine with addition with small zoom factor, and moved to other issues - and there's probably a more obvious way to handle that :)
Also, my frustration was there before this added support for SVG/_scale_factor_0 - so you can ignore this new stuff when you'll have a look.

Reply:
Keep in mind that I (very) recently tweaked it to at least be able to "consistently" hit the floor & ceiling values.

I'm possibly even more maths-impaired than @poire-z, and I definitely concur that it doesn't work as it ought to ;).